.github/workflows/pytest.yml

```diff
@@ -97,17 +97,17 @@ jobs:
           pip install --no-deps --editable .
           make pytest-unit
 
-      - name: Upload coverage
+      - name: Upload unit tests coverage artifact
         uses: actions/upload-artifact@v4
         with:
           name: coverage-unit
-          path: coverage.xml
+          path: ./*coverage*
 
   ci-integration:
     runs-on:
       group: large-runner-group
       labels: ubuntu-22.04-4core
-    if: github.event.merge_group || github.event.workflow_dispatch
+    if: github.event_name == 'workflow_dispatch' || github.event.merge_group
```

Comment:
Run integration tests (and thus coverage check) only on manual `workflow_dispatch` or if we are in the merge queue.

Reply:
Huh. Weird quirk of GHA I guess 👍